stackrox/stackrox
# ROX-18009: Delete non-postgres code in List UI

## Description

> deprecating our Postgres feature flag

Because this is the caboose, delete from src/types/featureFlags.ts file.

### Analysis

Find in Files 13 results in 11 files

| Folder                        | results | files |
| :---                          |    ---: |  ---: |
| Components/workflow           |       1 |     1 |
| Containers/VulnMgmt/Dashboard |       1 |     1 |
| Containers/VulnMgmt/List      |      10 |     8 |
| types                         |       1 |     1 |

1. Delete obsolete comment-out code for **Policies** column.

### Residue

1. Move components into container:
    * ImageTableCountLinks
    * ClusterTableCountLink
2. Investigate `entityType` property in table columns.

## Checklist

## Testing Performed

TL;DR more than half of decrease is in main.js probably because of queryService in utils and possible also ImageTableCountLinks in Components.

1. `yarn lint` in ui
2. `yarn build` in ui
    * `wc build/static/js/*.js`
        branch - master: -11584 = 11084108 - 11095656
        branch - master main.js: -6417 = 4849270 - 4855687
3. `yarn start` in ui

### Manual testing

1. Visit /main/vulnerability-management
    * See menu: **CVEs**.
    * See menu: **Application & Infrastructure**.
        ![dashboard](https://github.com/stackrox/stackrox/assets/11862657/1917279e-38b8-442c-83fc-08d6fb4b91cb)

2. Visit /main/vulnerability-management/image-cves
    * See expected count of CVEs.
        ![image-cves](https://github.com/stackrox/stackrox/assets/11862657/8b2d3217-929c-40dd-8890-0edaf0fc8fac)

3. Visit /main/vulnerability-management/node-cves
    * See expected count of CVEs.
        ![node-cves](https://github.com/stackrox/stackrox/assets/11862657/512ab75c-962c-4365-ad22-5d59f6d55097)

4. Visit /main/vulnerability-management/cluster-cves
    * See expected count of CVEs.
        ![cluster-cves](https://github.com/stackrox/stackrox/assets/11862657/7e0fe2e7-d346-446a-b3f8-efc8aede0ec2)

5. Visit /main/vulnerability-management/nodes
    * See links in column: **Node CVEs**.
        ![nodes](https://github.com/stackrox/stackrox/assets/11862657/bf2470bc-ff34-4f37-b53e-9f8fe7ba7d4d)

6. Visit /main/vulnerability-management/images
    * See links in column: **Entities**.
        ![images](https://github.com/stackrox/stackrox/assets/11862657/d0494d31-1037-4ec0-a093-0cade53622df)

7. Visit /main/vulnerability-management/clusters
    * See links in columns: **Image CVEs**, **Node CVEs**, **Platform CVEs**.
        ![clusters](https://github.com/stackrox/stackrox/assets/11862657/49ce76cc-751c-4520-b053-5c9975b2d4e6)

8. Visit /main/vulnerability-management/namespaces
    * See links in column: **Image CVEs**.
        ![namespaces](https://github.com/stackrox/stackrox/assets/11862657/df84bee7-35ff-4d65-8a6b-68f1f0d9a1f9)

9. Visit /main/vulnerability-management/deployments
    * See links in column: **Image CVEs**.
        ![deployments](https://github.com/stackrox/stackrox/assets/11862657/83e1da6e-6c36-4239-997d-ece2858cdbb9)

10. Visit /main/vulnerability-management/node-components
    * See links in column: **Node CVEs**.
        ![node-components](https://github.com/stackrox/stackrox/assets/11862657/24fae1d9-937c-4004-9d5a-57237e110336)

11. Visit /main/vulnerability-management/image-components
    * See links in column: **Image CVEs**.
        ![image-component](https://github.com/stackrox/stackrox/assets/11862657/4eda44d8-a374-44fd-976a-7bae38e2a8a3)

### Integration testing

1. `yarn cypress-open` in ui/apps/platform
    * vulnmanagement/dashboard.test.js
    * vulnmanagement/imageCves.test.js
    * vulnmanagement/nodeCves.test.js
    * vulnmanagement/clusterCves.test.js
    * vulnmanagement/nodes.test.js
    * vulnmanagement/images.test.js
    * vulnmanagement/clusters.test.js
    * vulnmanagement/namespaces.test.js
    * vulnmanagement/deployments.test.js
    * vulnmanagement/nodeComponents.test.js
    * vulnmanagement/imageComponents.test.js
